pkg/manager/impl/task_execution_manager.go

```diff
@@ -162,7 +162,7 @@ func (m *TaskExecutionManager) CreateTaskExecutionEvent(ctx context.Context, req
 	}
 
 	currentPhase := core.TaskExecution_Phase(core.TaskExecution_Phase_value[taskExecutionModel.Phase])
-	if common.IsTaskExecutionTerminal(currentPhase) {
+	if common.IsTaskExecutionTerminal(currentPhase) && !common.IsTaskExecutionTerminal(request.Event.Phase) {
```

Comment:
what is the purpose of this? do we send multiple terminal events?

Reply:
you are right, this will be handled by flytepropeller